Proofs/Experiments/AbsVCG.lean

```diff
@@ -0,0 +1,383 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Shilpi Goel
+
+Experimental: Use the Correctness module to prove that this program implements
+absolute value correctly. The objective of this exercise is to determine the
+"shape" of the lemmas that are needed for automation and/or to modify the
+definitions in Correctness, if needed.
+-/
+import Arm
+import Tactics.StepThms
+import Tactics.Sym
+import Correctness.Correctness
+
+namespace AbsVCG
+
+def program : Program :=
+  def_program
+   [
+    (0x4005d0#64, 0x2a0003e1#32), --  mov w1, w0
+    (0x4005d4#64, 0x131f7c00#32), --  asr w0, w0, #31
+    (0x4005d8#64, 0x0b000021#32), --  add w1, w1, w0
+    (0x4005dc#64, 0x4a000020#32), --  eor w0, w1, w0
+    (0x4005e0#64, 0xd65f03c0#32)] --  ret
+
+/-- `Abs` structure: Field `w0` refers to the 32-bit input to the program that
+  is expected to be in register `w0` in the initial state, and `state` refers
+  to the Arm machine state.
+
+  We club the input and the state together so that we can express the
+  post-condition of the program only in terms of this structure `Abs`; see
+  `abs_post`. -/
```

Comment:
```suggestion
/--
`Abs` structure: Field `w0` refers to the 32-bit input to the program that
is expected to be in register `w0` in the initial state, and `state` refers
to the Arm machine state.

We club the input and the state together so that we can express the
post-condition of the program only in terms of this structure `Abs`; see
`abs_post`.
-/
```

The lean4 style seems to be to move the comments into a separate line when the comment is multiline. This is useful, because life is too short to maintain comment indentation :)